docs/Image-PrePull-Feature-Enhancement-Instruction-Document/Instruction-Document.md

```diff
@@ -0,0 +1,157 @@
+
+API example for pre-pull mirror image：
+
+```
+apiVersion: operations.kubeedge.io/v1alpha1
+kind: ImagePrePullJob
+metadata:
+  name: imageprepull-example
+  labels:
+    description:ImagePrePullLabel
+spec:
+  imagePrePullTemplate：
+    images:
+      - image1
+      - image2
+    nodes：
+      - edgenode1
+      - edgenode2
+    checkItems:
+      - "disk"
+    failureTolerate: "0.3"
+    concurrency: 2
+    timeoutSeconds: 180
+    retryTimes: 1
+    
+```
+
+
+## 1. Preparation
+
+**Example：Nginx Demo**
+
+Nginx is a lightweight image that allows users to demonstrate it without any prerequisite environment. Nginx image will be uploaded to a private image repository in advance. Users can call the pre-pull function API from the cloud to pre-pull the Nginx image to edge nodes from the private image repository.
+
+**1）This example requires KubeEdge version to be v1.16.0 or above, and Kubernetes version to be v1.27.0 or above. The selected version is KubeEdge v1.16.0 and Kubernetes version is v1.27.3.**
+
+```
+[root@ke-cloud ~]# kubectl get node
+NAME            STATUS    ROLES                  AGE   VERSION
+cloud.kubeedge   Ready    control-plane,master   3d   v1.27.3
+edge.kubeedge    Ready    agent,edge             2d   v1.27.7-kubeedge-v1.16.0
+
+Note: The following verification will use the edge node edge.kubeedge. If you refer to this article for related verification, the configuration of the edge node name in subsequent steps needs to be changed according to your actual situation.
+```
+
+**2）Ensure that the K8s apiserver has the following configuration enabled**
+
+
+```
+      taskManager:
+        enable: true  // 由false修改为true
+```
+changes can be made by editing the file kubectl edit configmap cloudcore -n kubeedge with commands, and restarting the cloudcore component of the K8s apiserver.
+
+**3）Preparing Sample Code **
+
+yaml file example code
+```
+apiVersion: operations.kubeedge.io/v1alpha1
+kind: ImagePrePullJob
+metadata:
+  name: imageprepull-example
+spec:
+  imagePrePullTemplate:
+    concurrency: 1
+    failureTolerate: '0.1'
+    images:
+      - test:nginx
+    nodeNames:
+      - edge.kubeedge
+    imageSecrets: default/secret
+    retryTimes: 1
+    timeoutSeconds: 120
+
+```
+
+## 2. Prepare the image and Secret for the private image repository
+Here is a private image repository prepared for demonstration purposes using Alibaba Cloud's registry URL: registry.cn-hangzhou.aliyuncs.com. The demo space used is jilimoxing, and modifications may be necessary based on actual circumstances during the actual operation.
+
+**1）Pushing nginx into the private image repository**
+
+```
+[root@cloud ~]# docker tag nginx registry.cn-hangzhou.aliyuncs.com/jilimoxing/test:nginx
+[root@cloud crds~]# docker push registry.cn-hangzhou.aliyuncs.com/jilimoxing/test:nginx
+```
+
+**2）Create a Secret on the cloud**
```

Comment:
1) is not necessary.   "Secret is not a required field in ImagePrePullJob. If you need to prepull private image, you can generate a secret for it."